dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/JdbcEventStore.java

```diff
@@ -1270,4 +1421,72 @@ private boolean userHasAccess( SqlRowSet rowSet )
             throw new IllegalArgumentException( e );
         }
     }
+
+    private void populateCaches( EventSearchParams params, String sql, boolean isSuperUser )
+    {
+        IdSchemes idSchemes = ObjectUtils.firstNonNull( params.getIdSchemes(), new IdSchemes() );
+        IdScheme idScheme = idSchemes.getDataElementIdScheme();
+
+        if ( idScheme == IdScheme.ID || idScheme == IdScheme.UID )
+        {
+            return;
+        }
+
+        SqlRowSet rowSet = jdbcTemplate.queryForRowSet( sql );
+
+        Set<String> deUids = new HashSet<>();
+
+        while ( rowSet.next() )
+        {
+            if ( (params.getCategoryOptionCombo() == null && !isSuperUser && !userHasAccess( rowSet )) )
+            {
+                continue;
+            }
+
+            if ( !org.springframework.util.StringUtils.isEmpty( rowSet.getString( "psi_eventdatavalues" ) ) )
+            {
+                Set<EventDataValue> eventDataValues = convertEventDataValueJsonIntoSet( rowSet.getString( "psi_eventdatavalues" ) );
+                eventDataValues.forEach( edv -> deUids.add( edv.getDataElement() ) );
+            }
+        }
+
+        if ( idScheme.isAttribute() )
+        {
+            List<DataElement> dataElements = manager.get( DataElement.class, deUids );
+            dataElements.forEach( de -> dataElementToCodeCache.put( de.getUid(), de ) );
+        }
+        else
+        {
+            String dataElementsUidsSqlString = deUids.stream().collect( Collectors.joining( "', '", "'", "'" ) );
+
+            String deSql = "SELECT de.uid, av.value " +
+                "FROM dataelement de JOIN dataelementattributevalues deav ON de.dataelementid = deav.dataelementid " +
+                "JOIN attributevalue av ON av.attributevalueid = deav.attributevalueid " + "WHERE de.uid IN (" +
+                dataElementsUidsSqlString + ")";
+
+            SqlRowSet deRowSet = jdbcTemplate.queryForRowSet( deSql );
+
+            while ( deRowSet.next() )
+            {
+                dataElementToAttributeCache.put( deRowSet.getString( "uid" ), deRowSet.getString( "value" ) );
+            }
+        }
+    }
+
+    private void clearCaches()
```

Comment:
I don't we need this part.

Reply:
Sure. Not needed if it will be only within the method and not a class.